src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -1635,22 +1635,22 @@ public void Welcome(Address joinWith, UniqueAddress from, Gossip gossip)
         public void Leaving(Address address)
         {
             // only try to update if the node is available (in the member ring)
-            if (LatestGossip.Members.Any(m => m.Address.Equals(address) && m.Status is MemberStatus.Joining or MemberStatus.WeaklyUp or MemberStatus.Up))
+            foreach(var mem in LatestGossip.Members.Where(m => m.Address.Equals(address)))
```

Comment:
TLDR - iterate over each `Member` with the same `Address` and filter them one at a time - don't do what we did before, which is to check if there are members that match this condition and then try to state-transition them all at the same time. This is what creates the `IllegalOperation: invalid state transition` errors.